components/constellation/constellation.rs

```diff
@@ -1405,10 +1405,6 @@ impl<Message, LTF, STF> Constellation<Message, LTF, STF>
 
             let window_size = old_pipeline.and_then(|old_pipeline| old_pipeline.size);
 
-            if let Some(old_pipeline) = old_pipeline {
```

Comment:
However, for the cases in which the document is salvageable, I believe this is handled in the spec by ignoring timer tasks for documents which are not fully active in step 1 of https://html.spec.whatwg.org/multipage/webappapis.html#event-loop-processing-model . Therefore, when the document becomes inactive would be equivalent.